mava/wrappers/jumanji.py

```diff
@@ -217,3 +219,101 @@ def observation_spec(self) -> specs.Spec[Union[Observation, ObservationGlobalSta
             )
 
         return spec
+
+
+class CleanerWrapper(MultiAgentWrapper):
+    """Multi-agent wrapper for the Cleaner environment.
+
+    Do not use the AgentID wrapper with this env, it has implicit agent IDs.
+    """
+
+    def __init__(self, env: Cleaner, has_global_state: bool = False):
+        super().__init__(env)
+        self.has_global_state: bool = False
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for the Cleaner environment."""
+
+        def create_agents_view(grid: chex.Array, agents_locations: chex.Array) -> chex.Array:
+            agents_locations = agents_locations
+            num_agents = agents_locations.shape[0]
+            dirty_channel = jnp.tile(jnp.where(grid == DIRTY, 1, 0), (num_agents, 1, 1))
+            wall_channel = jnp.tile(jnp.where(grid == WALL, 1, 0), (num_agents, 1, 1))
+            xs, ys = agents_locations[:, 0], agents_locations[:, 1]
+            pos_per_agent = jnp.repeat(jnp.zeros_like(grid)[None, :, :], num_agents, axis=0)
+            pos_per_agent = pos_per_agent.at[jnp.arange(num_agents), xs, ys].set(1)
+            agents_pos = jnp.tile(jnp.sum(pos_per_agent, axis=0), (num_agents, 1, 1))
+
+            transformed = jnp.stack(
+                [dirty_channel, wall_channel, agents_pos, pos_per_agent], axis=-1
+            )
+            return transformed
+
+        def create_global_state(agents_view: chex.Array) -> chex.Array:
+            global_state = agents_view[..., :3]
+            return global_state
+
+        obs_data = {
+            "agents_view": create_agents_view(
+                timestep.observation.grid, timestep.observation.agents_locations
+            ),
+            "action_mask": timestep.observation.action_mask,
+            "step_count": jnp.repeat(timestep.observation.step_count, self._num_agents),
+        }
+        reward = jnp.repeat(timestep.reward, self._num_agents)
+        discount = jnp.repeat(timestep.discount, self._num_agents)
+        if self.has_global_state:
+            obs_data["global_state"] = create_global_state(timestep.observation.grid)
+            return timestep.replace(
+                observation=ObservationGlobalState(**obs_data), reward=reward, discount=discount
+            )
+        else:
+            return timestep.replace(
+                observation=Observation(**obs_data), reward=reward, discount=discount
+            )
```

Comment:
nit, newlines for readability
```suggestion
        obs_data = {
            "agents_view": create_agents_view(
                timestep.observation.grid, timestep.observation.agents_locations
            ),
            "action_mask": timestep.observation.action_mask,
            "step_count": jnp.repeat(timestep.observation.step_count, self._num_agents),
        }
        
        reward = jnp.repeat(timestep.reward, self._num_agents)
        discount = jnp.repeat(timestep.discount, self._num_agents)
        
        if self.has_global_state:
            obs_data["global_state"] = create_global_state(timestep.observation.grid)
            return timestep.replace(
                observation=ObservationGlobalState(**obs_data), reward=reward, discount=discount
            )
        else:
            return timestep.replace(
                observation=Observation(**obs_data), reward=reward, discount=discount
            )
```